### PR TITLE
[AMDGPU] llvm.amdgcn.raw.buffer.load.format intrinsic supports v4i32 as return type.

### PR DESCRIPTION
**Pull request by sstipano**
…4i32 as return type. This went unnoticed until CreateIntrinsic with implicit mangling was used.

### CODE REVIEW

File: llvm/include/llvm/IR/IntrinsicsAMDGPU.td [hunk 1]
```diff
@@ -1162,7 +1162,7 @@ class AMDGPURawBufferLoad<LLVMType data_ty = llvm_any_ty> : DefaultAttrsIntrinsi
                      //           all: volatile op (bit 31, stripped at lowering)
   [IntrReadMem, ImmArg<ArgIndex<3>>], "", [SDNPMemOperand]>,
   AMDGPURsrcIntrinsic<0>;
-def int_amdgcn_raw_buffer_load_format : AMDGPURawBufferLoad<llvm_anyfloat_ty>;
+def int_amdgcn_raw_buffer_load_format : AMDGPURawBufferLoad;
```
Comment (jayfoad) [inline llvm/include/llvm/IR/IntrinsicsAMDGPU.td:1165]: The data_ty template argument is unused now so please remove it.

Comment (sstipano) [reply llvm/include/llvm/IR/IntrinsicsAMDGPU.td:1165]: I saw unused data_ty argument in couple of places, that's why I left it. Now Removed.

**Review by jayfoad — APPROVED** [review]
LGTM.

**Comment by shiltian** [comment]
And I don't think LLPC is anything meaningful to the LLVM community

**Comment by arsenm** [comment]
> And I don't think LLPC is anything meaningful to the LLVM community

Don't state what it can do, state the intrinsic supports

**Comment by jayfoad** [comment]
> And I don't think LLPC is anything meaningful to the LLVM community

Well it's an open source frontend that generates LLVM IR: https://github.com/GPUOpen-Drivers/llpc

**Pull request merged**